pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines functions to decompose controlled operations
+"""
+
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.wires import Wires
+
+
+def ctrl_decomp_zyz(target_operation: Operator, control_wires: Wires):
+    """Decompose the controlled version of a target single-qubit operation
+
+    This function decomposes a controlled single-qubit target operation using the
+    decomposition defined in section 5 of
+    `Barenco et al. (1995) <https://arxiv.org/abs/quant-ph/9503016>`_.
+
+    .. warning:: This method will add a global phase for target operations that do not
+        belong to the SU(2) group.
+
+    Args:
+        target_operation (~.operation.Operator): the target operation to decompose
+        control_wires (~.wires.Wires): the control wires of the operation
+
+    Returns:
+        list[Operation]: the decomposed operations
+
+    Raises:
+        ValueError: if `target_operation` is not a single-qubit operation
+
+    **Example**
+
+    .. code-block:: python
+
+        # We can create a controlled operation using `qml.ctrl`, or by creating the
+        # decomposed controlled version of using `qml.ctrl_decomp_zyz`.
+        op = qml.RX(0.123, wires=2)
+        ctrl_op = qml.ctrl(op, [0, 1])
+        ctrl_decomp = qml.ops.ctrl_decomp_zyz(op, [0, 1])
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def expected_circuit():
+            for i in range(3):
+                qml.Hadamard(i)
+            qml.apply(ctrl_op)
```

Comment:
```suggestion
            qml.ctrl(op, [0,1])
```
Same thing here.  We can just call `ctrl` here instead of pre-initializing and then using `apply`.  This is a little more readable and more in line with how we tend to style things.